test/clone_schema.test.js

```diff
@@ -32,6 +56,8 @@ test('cloning schema with cloneSchema option', function (assert) {
     nullable: true
   }
 
+  var cloned = JSON.parse(JSON.stringify(schema))
```

Comment:
This looks like it’s reversing #9 which @XVincentX did to help circular references. Can we make sure we’re not speeding things up by breaking some edge cases? 😅

Reply:
The general problem is that this library is mutating the passed object; the `cloneSchema` thing happens exclusively in case you want immutability from the outside so that the original object stays the same. It's a band-aid solution.

The best solution here would be to simply stop mutating the object in place using the fp counterpart of lodash (such as https://gist.github.com/brauliodiez/9fd567fdc8b2695c11a61daa50475f43), but it would require a significant change in this library (and likely a breaking change as well)